src/features/chart/redux/middleware.ts

```diff
@@ -1,10 +1,17 @@
 import { BOOTSTRAP } from "../../../store/constants";
 import { UPDATE_CHART_RANGE } from "./constants";
-import { updateChartDataAction } from "./actions";
+import { updateChartDataAction, ChartData } from "./actions";
+import { SocketService } from "../../../services";
+import { Middleware } from "redux";
+import { GlobalState } from "../../../store";
+
+type Dependancies = {
+  socketService: SocketService;
+};
 
 export const chartMiddleware = ({
   socketService
-}) => store => next => action => {
+}: Dependancies): Middleware<{}, GlobalState> => store => next => action => {
```

Comment:
Maybe better to call it AppState as GlobalState has already installed in js

Reply:
The term `Global` has connotations to JS's `global`, which is probably best avoided.

We refer to this as a "Web Application" ("WebApp"), so `AppState` tends to be the term we use

Reply:
Good idea both - will change

Reply:
My fault - it's called `GlobalState` in my current project. `AppState` is a better name though